ios/brave-ios/Sources/Brave/Frontend/Browser/LinkPreviewViewController.swift

```diff
@@ -32,21 +32,21 @@ class LinkPreviewViewController: UIViewController {
       return
     }
 
-    currentTab = TabState(
-      configuration: parentTab.configuration,
-      type: parentTab.isPrivate ? .private : .regular
-    ).then {
-      $0.miscDelegate = browserController
-      $0.createWebview()
-      $0.addPolicyDecider(browserController)
-      $0.delegate = browserController
-      $0.downloadDelegate = browserController
-      $0.webScrollView?.layer.masksToBounds = true
-    }
+    let tab = TabStateFactory.create(
+      with: .init(
+        configurationProvider: { parentTab.configuration },
```

Comment:
Does it have to be a lambda? Also it captures `parentTab` strongly is it okay? If the parent tab dies, will it still be held by this tab IE: via `window.open`?

Reply:
The closure is non-escaping so no issues with capturing Tab here, though I dont know that it needs to be a closure at all.. will give it another once over